curriculum/challenges/english/14-responsive-web-design-22/learn-html-forms-by-building-a-registration-form/63541ef4f96cd82e8e6c788a.md

```diff
@@ -0,0 +1,52 @@
+---
+id: 63541ef4f96cd82e8e6c788a
+title: Step 12
+challengeType: 0
+dashedName: step-12
+---
+
+# --description--
+
+The `method` attribute specifies how to send form-data to page specified in the `action` attribute. The form-data can be sent as URL variables (with `method="get"`) or as HTTP post transaction (with `method="post"`).
+
+Add `post` value to `method` attribute in `form`.
+
+# --hints--
+
```

Comment:
```suggestion

You should have one `form` with `method` attribute

```

Then add the following test 

\`\`\`js
assert.equal(document.querySelectorAll('form')?.getAttribute('method')?.length, 1);
\`\`\`

Reply:
Maybe a third test, that checks for the amount of forms 🤔 

These test usually have a hint of the following.

> You shouldn't add a new `form` element.

Reply:
`assert.equal(document.querySelectorAll('form')?.getAttribute('method')?.length, 1);` -> return `Node` element 
Calling `getAttribute()` is not possible.. !! 
Should we iterate over each `form` element and validate ?
_Suggestion_:  Just validating for one `form` will be sufficient since validation for multiple form element will be done as first test !!

Reply:
> Suggestion: Just validating for one form will be sufficient since validation if multiple form element are present or not will be done as first test !!

Sounds good go ahead 👍, I am happy as long as it isn't possible for user to add the code below

> ```html
> <form method='post'></form>
> <form action='https://register-demo.freecodecamp.org'></form>
> ```

Reply:
- [x] Added test case to check if multiple forms are added 
- [x] Added test case to check if method value is present or not